otter/util/pure_http.py

```diff
@@ -27,18 +28,20 @@ class Request(object):
 
     treq = logging_treq
 
-    @inlineCallbacks
-    def perform_effect(self, dispatcher):
-        """
-        Perform the request with treq.
 
-        :return: A two-tuple of (HTTP Response, content as bytes)
-        """
-        response = yield self.treq.request(self.method.upper(), self.url,
-                                           headers=self.headers,
-                                           data=self.data, log=self.log)
-        content = yield self.treq.content(response)
-        returnValue((response, content))
+@deferred_performer
+@inlineCallbacks
+def perform_request(dispatcher, intent):
+    """
+    Perform the request with treq.
+
+    :return: A two-tuple of (HTTP Response, content as bytes)
+    """
+    response = yield intent.treq.request(intent.method.upper(), intent.url,
```

Comment:
You're right, it definitely shouldn't, and it was a known compromise from the very beginning. But I think that should be resolved in another PR, or in fxhttp, when we (tomprince and I and whoever else wants to help) implement that. I don't mind this code-smell for a little bit longer :)